eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/Onion.scala

```diff
@@ -123,63 +124,76 @@ Notes:
 case class OnionRoutingPacket(version: Int, publicKey: ByteVector, payload: ByteVector, hmac: ByteVector32)
 
 /** Tlv types used inside onion messages. */
-sealed trait OnionTlv extends Tlv
+sealed trait OnionPaymentPayloadTlv extends Tlv
 
-object OnionTlv {
+sealed trait OnionMessagePayloadTlv extends Tlv
+
+sealed trait BlindedTlv extends Tlv
```

Comment:
We should really stay very close to spec names in our tlv traits, otherwise it's really hard to verify we correctly follow the spec. I need to spend time on the spec PR to see if that name needs to be clarified at the spec level or not.